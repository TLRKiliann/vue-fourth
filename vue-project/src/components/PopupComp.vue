<script lang="ts">
import { defineComponent } from 'vue';
export default defineComponent({
  name: 'PopupComp',
  data() {
    return {
      showPopup: false
    }
  },
  methods: {},
})
</script>

<template>
    <h2 v-if="showPopup">My popup</h2>
    <h2 v-else="showPopup = false">Hide</h2>
    <button @click="showPopup = true">Show btn popup</button>
</template>

<style scoped>

</style>