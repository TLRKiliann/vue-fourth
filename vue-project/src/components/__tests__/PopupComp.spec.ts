import { describe, it, expect } from 'vitest'

import { mount } from '@vue/test-utils'
import PopupComp from '../PopupComp.vue'

describe('PopupComp', () => {
  it('snapshot UI testing', () => {
    const wrapper = mount(PopupComp,{});
    expect(wrapper.text()).toMatchSnapshot()
  })
})