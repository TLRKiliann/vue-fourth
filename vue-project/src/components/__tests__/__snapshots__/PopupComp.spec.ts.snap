// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PopupComp > snapshot UI testing 1`] = `"My popup"`;
